ldapsearch crashes with invalid prameters (ldapsearch -w password -W)

Upstream ITS: #6849
Upstream commit: d7a8eff
Resolves: #743781

diff --git a/clients/tools/common.c b/clients/tools/common.c
index da14c60..262631e 100644
--- a/clients/tools/common.c
+++ b/clients/tools/common.c
@@ -1067,6 +1067,12 @@ tool_args( int argc, char **argv )
 		}
 #endif
 	}
+
+	if ( ( pw_file || want_bindpw ) && !BER_BVISNULL( &passwd ) ) {
+		fprintf( stderr, "%s: -%c incompatible with -w\n",
+			prog, ( pw_file ? 'y' : 'W' ) );
+		exit( EXIT_FAILURE );
+	}
 }
 
 
